### PR TITLE
Improvement/merge core

### PR DESCRIPTION
**Pull request by rfreitas**
The end of core is near...

#### Merging of core

I have added sdk-core as an origin on my local. I then created a branch from it and merged development into it. This was done in order to preserve the history of onfido/onfido-sdk-core

I resolved conflicts by:
1.  Removing files specific to sdk core repo
2. Moved onfido-sdk-core source files to `src/core`
3. Changed the imports to point to this `src/core` instead of the npm core
4. Removed the testing framework, which is not being used, since this was causing an error with no quick fix

#### Cleanup

Some dependencies were moved to their correct place, namely babel-runtime and pdfObject.

#### Fix for the annoying `resolve failed:  { Error: Cannot find module ...`

Changed the babel preset to `es2015` from `es2015-minimal`, see: https://github.com/developit/preact-boilerplate/issues/88

#### Linter not working on Atom

Added `node_modules` to lint ignore

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://169-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://169-pr-onfido-sdk-ui-onfido.surge.sh

File: package.json [hunk 1]
```diff
@@ -95,13 +80,13 @@
     "webpack-dev-server": "^1.14.1"
   },
   "dependencies": {
+    "pdfobject": "2.0.201604172",
```
Comment (rfreitas) [inline package.json]: @stefaniacardenas just headsup, runtime dependencies should not go under dev dependencies ;)

Comment (stefaniacardenas) [inline package.json]: Oops 😶

**Comment by rfreitas** [comment]
Travis automatic deployment: https://169-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
@hjheath CX-985 has been created

**Comment by rfreitas** [comment]
Travis automatic deployment: https://169-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://169-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://169-pr-onfido-sdk-ui-onfido.surge.sh

File: .babelrc [hunk 2]
```diff
@@ -1,13 +1,19 @@
 {
   "sourceMaps": true,
   "presets": [
-    "es2015-minimal",
+    "es2015",
```
Comment (rfreitas) [inline .babelrc]: @stephencookdev if you remember asking me about the errors of  `resolve failed:` this is what was causing it.

Comment (rfreitas) [inline .babelrc]: @hjheath this adds 2% to the bundle size.

File: .babelrc [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
     "stage-0"
   ],
   "plugins": [
     ["transform-decorators-legacy"],
     ["add-module-exports"],
     ["transform-es2015-modules-umd"],
-    ["transform-react-jsx", { "pragma": "h" }]
+    ["transform-react-jsx", { "pragma": "h" }],
+    ["transform-runtime", {
+      "helpers": false,
```
Comment (rfreitas) [inline .babelrc]: @hjheath without these extra options the bundle size increases 2.5%

File: src/core/core/events.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import EventEmitter from 'eventemitter2'
```
Comment (stefaniacardenas) [inline src/core/core/events.js]: As we discussed, having a folder called core inside core isn't very meaningful, shall we handle this as part of the story in which we remove the unused code?

Comment (rfreitas) [inline src/core/core/events.js]: outside of the scope, the goal was to merge and try to preserve history, not to remove dead code. However, I have added an issue for it.

**Pull request merged**